### DISCUSSION TITLE
trailing commas on dictionaries should be all or none

### DISCUSSION THREAD

**Discussion by docwhat** [terraform-linters/tflint#1541] [Ideas]
## Introduction

Uneven commas can make terraform code confusing.  On dictionaries, they should either be on all or none.

I would expect that `tflint` would raise an warning on this snippet of code:

```tf
locals {
  a_dictionary = {
    "one"  = "fish"
    "two"  = "fish",
    "red"  = "fish"
    "blue" = "fish"
  }
}
```

I would expect the `two` line to be flagged as the rest of the lines do not have commas.

## Proposal

The rule should take these values:

- `all`: All dictionary entries should end in a comma.
- `none`: All dictionary entries should not end in a comma.
- `match`: All the dictionary entries in a given dictionary should be the same; either all commas or all without commas.

## References

None

**Comment by bendrucker** [comment]
At this time the `terraform` ruleset covers official recommendations from the Terraform docs. Outside of these official recommendations, we are not accepting style enforcement rules into any official rulesets in this org. See https://github.com/terraform-linters/tflint-ruleset-terraform/issues/22 for discussion.

You're of course free to implement your own ruleset plugin. If you publish one, it would be great if you'd share the URL here in case others are looking for a similar rule.

**Comment by Gaardsholt** [comment]
@docwhat I don't know if you still need this, but I have added it to my plugin, as I needed it myself :)
The rule isn't enabled by default, so you will have to explicitly enable it.

```hcl
rule "terraform_map_trailing_comma" {
  enabled = true
  style   = "none"   # Options: "all", "none", "match" (default)
}
```
